dataland-frontend/src/components/forms/CreateEuTaxonomyForFinancials.vue

```diff
@@ -201,70 +201,45 @@
                   </div>
                 </div>
 
-                <div
-                  v-for="financialServiceOption of confirmedSelectedFinancialServiceOptions"
```

Comment:
This was the key issue. Due to Asset Management section being shaped differently, it would cause an error in the form model when being deleted from the kpi sections. That's why (below) I moved the kpi sections into their own scope `name="kpiSectionsModel"` ... (After your comments above, I removed the `v-model=kpiSections"` code)